### PR TITLE
Only log when config actually changes.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @toregge PR

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/memoryflush.h [hunk 1]
```diff
@@ -34,6 +34,14 @@ class MemoryFlush : public IFlushStrategy
                uint64_t maxMemoryGain_in,
                double diskBloatFactor_in,
                vespalib::duration maxTimeGain_in);
+        bool equal(const Config & rhs) const {
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/memoryflush.h:37]: What is the rationale of not using _operator==_ instead ?

**Review by geirst — APPROVED** [review]

File: searchcore/src/vespa/searchcore/proton/server/memoryflush.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/memoryflush.h:37]: Probably too much java lately.

Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/memoryflush.h:39]: Fixed

File: searchcore/src/tests/proton/server/memory_flush_config_updater/memory_flush_config_updater_test.cpp [hunk 2]
```diff
@@ -75,6 +75,55 @@ TEST_F("require that strategy is updated when setting new config", Fixture)
...
+    EXPECT_FALSE(a == b);
+    EXPECT_TRUE( a != b);
+    EXPECT_FALSE(b.equal(a));
+    EXPECT_FALSE(b == a);
+    EXPECT_TRUE( b != a);
+}
+
+TEST("require that MemoryFlush::Config equal is correct") {
+    MemoryFlush::Config a, b;
+    expectEqual(a, b);;
```
Comment (geirst) [inline searchcore/src/tests/proton/server/memory_flush_config_updater/memory_flush_config_updater_test.cpp:100]: Nitpick: Some extra ";" have been added. Applies further down as well.

Comment (baldersheim) [reply searchcore/src/tests/proton/server/memory_flush_config_updater/memory_flush_config_updater_test.cpp:100]: Fixed in #18387

**Pull request merged**